test/integration/scheduler/queue_test.go

```diff
@@ -152,6 +152,65 @@ func TestSchedulingGates(t *testing.T) {
 	}
 }
 
+// We create a gated and non-gated pod. After scheduling and deleting the
+// non-gated pod, we ensure that the gated pod is schedulable after the removal
+// of its gate.
+func TestGatedPodSchedulableAfterEvent(t *testing.T) {
```

Comment:
Integration test in this file can only cover functionality instead of performance, so I'm not sure it's worth adding a _duplicate_ integration test here as the same function path has been covered in `TestSchedulingGates`.

Reply:
Oh, if there is already a test covering it, then I'm fine removing this.

Reply:
yup, there was one; otherwise, we cannot promote this feature even for alpha...

Reply:
There is a difference: `TestGatedPodSchedulableAfterEvent` covers the early-skip path added in this PR. `TestSchedulingGates` doesn't hit this path, as the only event is a pod update. Do you think it is worth keeping the new test?

Reply:
Maybe we can enhance the existing test to add the additional pod termination?

Reply:
yup, feel free to refactor `TestSchedulingGates` to add this as a sub-test.